source/android/adaptivecards/src/androidTest/java/io/adaptivecards/objectmodel/DateInputPropertiesTest.java

```diff
@@ -194,6 +199,103 @@ public DateInput castTo(BaseCardElement element) {
         }
     }
 
+    @Test
+    public void MinValidationTest() throws Exception
+    {
+        DateInput dateInput = TestUtil.createMockDateInput();
+        dateInput.SetMin(c_inRangeDateValues[0]);
+
+        DateInputHandler dateInputHandler = new DateInputHandler(dateInput, null);
+        dateInputHandler.setView(new EditText(InstrumentationRegistry.getContext()));
+
+        // Validate that empty input is always valid
+        Assert.assertEquals(true, dateInputHandler.isValid());
```

Comment:
We really should try to split these out into different tests just because the behavior with empty input etc is confusing and poorly documented anyways; At least having tests that are very specific and testing specific behavior can help drive some clarity for future maintainers including yourself :)